mne/preprocessing/optical_density.py

```diff
@@ -0,0 +1,36 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+from ..io import BaseRaw
+from ..io.constants import FIFF
+from ..utils import _validate_type
+from ..io.pick import _picks_to_idx
+
+from numpy import mean, divide, log, ones
+
+
+def optical_density(raw):
+    r"""Convert NIRS raw data to optical density.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+
+    Returns
+    -------
+    raw : instance of Raw
+        The modified raw instance. Operates in place.
+
+    """
+    _validate_type(raw, BaseRaw, 'raw')
+    picks = _picks_to_idx(raw.info, 'fnirs_raw')
+    data_means = mean(raw.get_data(), 1)
+    for ii in picks:
+        data = -1.0 * log(divide(raw.get_data(ii).T,
+                          ones((len(raw), 1)) * data_means[ii]))
+        raw._data[ii, :] = data.T
```

Comment:
```suggestion
        raw._data[ii] /= means[ii]
        np.log(raw._data[ii], out=raw._data[ii])
```
And somehow deal with cases where `means[ii]` is zero, like:
```
mask = (means[picks] == 0)
means[mask] = 1.
for ii in picks:
    ...
raw._data[picks[mask]] = 0.